cluster-autoscaler/utils.go

```diff
@@ -271,3 +271,15 @@ func fixNodeGroupSize(context *AutoscalingContext, currentTime time.Time) (bool,
 	}
 	return fixed, nil
 }
+
+// Enables cloud provider specific node protection for each node group.
+func protectNodes(context *AutoscalingContext) error {
+	for _, group := range context.CloudProvider.NodeGroups() {
```

Comment:
Makes sense to me. I'd have to then pass in the instance protection flag when we build the aws cloud provider since the instance protection flag only applies for aws. I'll wait for https://github.com/kubernetes/contrib/pull/2226 to be merged so we don't get more conflicts.